[aoti] skip input symbol codegen for sympy expr w/ many symbols

ColinPeppler opened the pull request:
Issue was that
- symbol-ids appeared out-of-order w.r.t to the order of the forward inputs 
```
def forward(arg0 # [(s3 - 1) + s4, 32], arg1 #[(s3 - 1)] ..)
```
- this causes codegen to fail because it expects all the base symbols `s4,s3` to have been codegen-ed already.
- well, we can skip codegen-ing sympy expr with many symbols e.g. `(s3 - 1) + s4` because `s3` and `s4` will be codegen-ed by other inputs.

```
# for example
s3 = arg1.size(0) + 1
s4 = argN.size(0)
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152579



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler committed:
[aoti] skip codegen for sympy expr when codegening input

[ghstack-poisoned]

ColinPeppler committed:
Update on "[aoti] skip codegen for sympy expr when codegening input"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

jingsh commented on test/inductor/tmp.txt:
remove this file from commit

ColinPeppler committed:
Update on "[aoti] skip codegen for sympy expr when codegening input"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler committed:
Update on "[aoti] skip codegen for sympy expr when codegening input"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler committed:
Update on "[aoti] skip input symbol codegen for sympy expr w/ many symbols"

Issue was that
- symbol-ids appeared out-of-order w.r.t to the order of the forward inputs 
```
def forward(arg0 # [(s3 - 1) + s4, 32], arg1 #[(s3 - 1)] ..)
```
- this causes codegen to fail because it expects all the base symbols `s4,s3` to have been codegen-ed already.
- well, we can skip codegen-ing sympy expr with many symbols e.g. `(s3 - 1) + s4` because `s3` and `s4` will be codegen-ed by other inputs.

```
# for example
s3 = arg1.size(0) + 1
s4 = argN.size(0)
```




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented on test/inductor/test_aot_inductor.py:
```diff
@@ -4435,6 +4435,42 @@ def forward(self, x):
+
+        dim = torch.export.Dim("dimensionality")
+        derived_dim = 2 * dim
+        spec = {
+            "getitem_54": (Dim.AUTO,),
+            "getitem_52": (Dim.AUTO,),
+            "getitem_19": (derived_dim,),
+            "values_2": (Dim.AUTO,),
+            "offsets": (Dim.AUTO,),
+        }
```
this gives us this setup.

```
arg0_1: "[s33 + 2*s40 + 1]"
arg1_1: "[s33 + 2*s40 + 1]"
arg2_1: "[2*s40]"
arg3_1: "[s33 + 1]"
arg4_1: "[s33]"
```

laithsakka replied:
shall we add this as a comment?

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -4435,6 +4435,42 @@ def forward(self, x):
+            "offsets": (Dim.AUTO,),
+        }
+
+        result = AOTIRunnerUtil.run(model, inps, dynamic_shapes=spec)
```
Why not using `self.check_model`?

ColinPeppler committed:
Update on "[aoti] skip input symbol codegen for sympy expr w/ many symbols"

Issue was that
- symbol-ids appeared out-of-order w.r.t to the order of the forward inputs 
```
def forward(arg0 # [(s3 - 1) + s4, 32], arg1 #[(s3 - 1)] ..)
```
- this causes codegen to fail because it expects all the base symbols `s4,s3` to have been codegen-ed already.
- well, we can skip codegen-ing sympy expr with many symbols e.g. `(s3 - 1) + s4` because `s3` and `s4` will be codegen-ed by other inputs.

```
# for example
s3 = arg1.size(0) + 1
s4 = argN.size(0)
```




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented:
@pytorchbot merge
